src/core/include/openvino/runtime/tensor.hpp

```diff
@@ -37,6 +37,7 @@ class OPENVINO_API Tensor {
 protected:
     std::shared_ptr<InferenceEngine::Blob> _impl;  //!< Shared pointer to internal tensor representation
     std::shared_ptr<void> _so;                     //!< Reference to dynamically loaded library
+    std::shared_ptr<void> _addSo;                  //!< Reference to additional loaded library in virual plugin
```

Comment:
What about the tensor is used by multiple levels plugins? 
How about change the _so to vector?

Reply:
> What about the tensor is used by multiple levels plugins? How about change the _so to vector instead of having separate variable?

I summarize the problem scenarios as these, based on which I think _so is enough to cover the scenarios:
1) real hw device + batch enabled + network not batchable, in this case, the only needed so is the real hw so
2) virtual device as auto/multi, with only one hw device in selection list, like AUTO:GPU or MULTI:GPU, in this case, we also only need the real hw so. (for auto/multi with >=2 target devices, we are allocating own buffers and thus not touching hw buffers directly)
3) virtual device as hetero, since the structures of network in hetero plugin is decided in the runtime, we also query the needed so based on the real infer request which is handling the original input/outputs

Reply:
Actually as I know there is a future plan to have some another virtual device which will use hetero plugin.
(e.g., NewVirtualPlugin=>HeteroPlugin=>[GPUPluign+XXXPlugin]) 

Also my question was not only for the functional impact but also the variable management. 
so + add_so seems somewhat confusing. 
So I wanted to ask whether it is possilbe to let the tensor have vector<shared_ptr> so, so that it can hold any required targets.

Reply:
ahh.. yes thanks for the good suggestion, I'll update

Reply:
manage the tensor life cycle using vector of dependecy targets.
Actually there are remote context and variable states needs to implement as well, I would suggest use new jira tickets to track these seperately.

Reply:
You mean it will require too many changes for adding  SO  deps to vector? 
Yeah it is okay to deal with that kind of refactoring as a separate ticket and follow up there.

Reply:
yes, this will involve more changes in seperate files, but again, the changes are minor from functionality point of view.

Reply:
Okay, lets go with this solution for now and refactor this after CF.